src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -75,7 +75,7 @@ public override void RequirementCheck(AgentTaskPluginExecutionContext executionC
         }
     }
 
-    public sealed class BitbucketGitSourceProvider : AuthenticatedGitSourceProvider
+    public class BitbucketGitSourceProvider : AuthenticatedGitSourceProvider
```

Comment:
I am going through this sequentially, do we need to derive from these classes? Feels like a code smell.